File: backend/src/main/java/com/vocabpet/backend/entity/Vocabulary.java
package com.vocabpet.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

import com.vocabpet.backend.entity.enums.Difficulty;
import com.vocabpet.backend.entity.enums.PartOfSpeech;

@Entity
@Table(name = "vocabularies")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vocabulary {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String word;

    @Column(columnDefinition = "TEXT")
    private String meaning;

    @Column(columnDefinition = "TEXT")
    private String example;

    @Enumerated(EnumType.STRING)
    private Difficulty difficulty;

    @Enumerated(EnumType.STRING)
    private PartOfSpeech partOfSpeech;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "deck_id", nullable = false)
    private Deck deck;

    @Builder.Default
    @OneToMany(mappedBy = "vocabulary", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<UserVocabularyProgress> progresses = new ArrayList<>();
}

File: vocapet-desktop-friend/src/types/vocabulary.ts
export type Difficulty =
  | "EASY"
  | "MEDIUM"
  | "HARD";

export type PartOfSpeech =
  | "NOUN"
  | "VERB"
  | "ADJECTIVE"
  | "ADVERB";


export interface Vocabulary {
  id: number;

  word: string;

  meaning: string;

  example: string;

  difficulty: Difficulty;

  partOfSpeech: PartOfSpeech;

  deckId: number;

  bookmarked: boolean;
}